src/plugins/intel_cpu/src/emitters/x64/jit_snippets_emitters.cpp

```diff
@@ -783,59 +757,23 @@ BrgemmEmitter::BrgemmEmitter(jit_generator* h, cpu_isa_t isa, const ExpressionPt
     m_with_comp = brgemm_node->is_with_compensations();
     m_with_scratch = brgemm_node->is_with_scratchpad();
 
-    m_N_blk = brgemm_node->get_n_block_size();
-    m_K_blk = brgemm_node->get_k_block_size();
-    m_N_tail = m_N % m_N_blk;
-    m_K_tail = m_K % m_K_blk;
-
-    m_N_blk_loop = m_N >= 2 * m_N_blk;
-    m_K_blk_loop = m_K >= 3 * m_K_blk;
-    OPENVINO_ASSERT((!brgemm_node->is_with_data_repacking()) || (!m_N_blk_loop && !m_K_blk_loop),
-                    "BrgemmEmitter doesn't support blocking by K, N dimensions when data repacking is needed!");
-
-    auto N = [&](size_t n) {
-        switch (n) {
-            case 0: return m_N_blk;
-            case 1: return m_N_tail;
-            default: OPENVINO_THROW("BrgemmEmitter detected unsupported N value");
-        }
-    };
-    auto K = [&](size_t k) {
-        switch (k) {
-            case 0: return m_K_blk;
-            case 1: return m_K >= 2 * m_K_blk ? m_K_blk : 0;
-            case 2: return m_K_tail;
-            default:  IE_THROW() << "BrgemmEmitter detected unsupported K value";
-        }
-    };
-
-    bool has_K_kernel = false;
-    for (size_t k = 0; k < BRGEMM_K_KERNEL_NUM; k++) {
-        bool has_N_kernel = false;
-        for (size_t n = 0; n < BRGEMM_N_KERNEL_NUM; n++) {
-            const size_t kernel_idx = getBrgIdx(k, n);
-            auto& brgemmCtx = m_brgCtxs[kernel_idx];
-
-            brgemmCtx.M = m_M;
-            brgemmCtx.N = N(n);
-            brgemmCtx.K = K(k);
-            brgemmCtx.LDA = leading_dimensions[0];
-            brgemmCtx.LDB = brgemm_node->is_with_data_repacking() ? rnd_up(m_N, brgemm_copy->get_n_block_size()) : leading_dimensions[1];
-            brgemmCtx.LDC = leading_dimensions[2];
-            brgemmCtx.dt_in0 = static_cast<dnnl_data_type_t>(DnnlExtensionUtils::IEPrecisionToDataType(brg0Prc));
-            brgemmCtx.dt_in1 = static_cast<dnnl_data_type_t>(DnnlExtensionUtils::IEPrecisionToDataType(brg1Prc));
-            brgemmCtx.beta = has_K_kernel ? 1 : 0;
-
-            if (brgemmCtx.N == 0 || brgemmCtx.N > m_N ||
-                brgemmCtx.K == 0 || brgemmCtx.K > m_K)
-                continue;
-
-            initBrgemm(brgemmCtx, m_brgKernels[kernel_idx], brgWithAMX);
-            has_N_kernel = true;
-        }
-        if (has_N_kernel)
-            has_K_kernel = true;
+    m_brgCtx.M = m_M;
+    m_brgCtx.N = m_N;
+    m_brgCtx.K = m_K;
+    m_brgCtx.LDA = leading_dimensions[0];
+    if (brgemm_node->is_with_data_repacking()) {
+        const auto brgemm_copy = brgemm_node->get_brgemm_copy();
+        const auto allocated_shape = brgemm_copy->get_data_repacking_shape(expr->get_input_port_descriptor(1)->get_shape());
+        m_brgCtx.LDB = *allocated_shape.rbegin();
```

Comment:
Unfortunately we can't use subtensor of this input. `brgemm_copy_b` repacking shape rounds N dimension for I8 Brgemms. For example, if N=68, this value is stored in the corresponding subtensor value. However, the rounded value is 128 - and exactly this value must be assigned to `m_brgCtx.LDB`
Of course, it seems strange, and we need to have valid subtensors on 1st input in case of data repacking, but I would prefer to remove this WA within BF16/I8 blocking support